scene/3d/physics_body_3d.cpp

```diff
@@ -1427,7 +1427,7 @@ void CharacterBody3D::_move_and_slide_grounded(double p_delta, bool p_was_on_flo
 					const PhysicsServer3D::MotionCollision &collision = result.collisions[0];
 
 					Vector3 slide_motion = result.remainder.slide(collision.normal);
-					if (collision_state.floor && !collision_state.wall && !motion_slide_up.is_zero_approx()) {
+					if (collision_state.floor && !collision_state.wall && !motion_slide_up.is_zero_approx() && !sliding_enabled) {
```

Comment:
This condition seems impossible now, because the encompassing check ensures that either slide enabled is true, or collision state floor is false.

Reply:
I see- I'll try to rework it. I think it messes up the sliding value in that block of code where it assigns to slide_motion.

Reply:
Maybe it does, I'm no expert on physics, but this change makes the block to never trigger, even when it should. So this is definitely not a fix.

Reply:
Ok, made an edit. It will correctly slide now if sliding is enabled (stop on slope is false), and that block of code will still run for cases where you want to stop of slope.